### ISSUE TITLE
Experiments are not exacty reproducable.

### ISSUE THREAD

**Issue by Abdul-Mukit** [roboflow/rf-detr#742]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

When using the exact same config and dataset, consecutive training runs result in very slightly changed mAP and loss outputs. I believe experiments should be exactly reproducible, given a fixed dataset and config. While trying to address https://github.com/roboflow/rf-detr/issues/262, this issue surfaced. 

### Things I tried but didn't work:
* Add the missing `TrainConfig.seed` in config.py.  

* Replaced the seed logics in https://github.com/roboflow/rf-detr/blob/ccbc5aa9db04b61efdd015446aab9a962e9a8801/src/rfdetr/main.py#L191 with https://github.com/roboflow/rf-detr/blob/ccbc5aa9db04b61efdd015446aab9a962e9a8801/src/rfdetr/util/utils.py#L143

* Removed https://github.com/roboflow/rf-detr/blob/ccbc5aa9db04b61efdd015446aab9a962e9a8801/src/rfdetr/engine.py#L158

* Set aug_config={}
I noticed that https://albumentations.ai/docs/4-advanced-guides/reproducibility/ suggests setting a seed, which I think is not being done in `transforms.py`.

### Environment

- rfdetr 1.5.0
- python 3.10
- torch 2.8.0
- torchvision 0.23.0

### Minimal Reproducible Example

Consecutive runs of:  

```
model = RFDETRSegSmall()

model.train(
    dataset_dir=dataset_dir,
    epochs=2,
    batch_size=1,
    grad_accum_steps=1,
    lr=1e-4,
    num_workers=2,
    device='cuda',
    output_dir=output_dir,
    segmentation_head=True,
    run_test=False,
    tensorboard=True,
    aug_config={}
)
```

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

**Comment by isaacrob** [comment]
Backprop through grid sample is non deterministic due to machine precision on accumulators and race conditions in the GPU. No run with deformable attention will be perfectly reproducible. See https://discuss.pytorch.org/t/f-grid-sample-non-deterministic-backward-results/27566 variance may be small but I'm mentioning it to say that the goal of perfect reproduction isn't achievable.

**Comment by Abdul-Mukit** [comment]
Got it. Thank you.
